test/widgets/action_sheet_test.dart

```diff
@@ -1085,6 +1086,139 @@ void main() {
       });
     });
   });
+
+  group('channel action sheet', () {
+    late ZulipStream someChannel;
+    late PerAccountStore store;
+
+    Future<void> prepare({UnreadMessagesSnapshot? unreadMsgs}) async {
+    final stream = eg.stream();
+    someChannel = stream;
+    addTearDown(testBinding.reset);
+
+    unreadMsgs ??= eg.unreadMsgs(channels: [
+      eg.unreadChannelMsgs(
+        streamId: stream.streamId,
+        topic: 'topic',
+        unreadMessageIds: [1],
+      ),
+    ]);
+
+    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+      realmUsers: [eg.selfUser, eg.otherUser],
+      streams: [someChannel],
+      subscriptions: [eg.subscription(someChannel)],
+      unreadMsgs: unreadMsgs));
+    store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+    connection = store.connection as FakeApiConnection;
+  }
+
+    Future<void> showFromInbox(WidgetTester tester) async {
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: const HomePage()));
+      await tester.pumpAndSettle();
+      check(find.byType(InboxPageBody)).findsOne();
+
+      await tester.pump();
+      await tester.longPress(find.text(someChannel.name).hitTestable());
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    Future<void> showFromSubscriptionList(WidgetTester tester) async {
+      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+        child: const SubscriptionListPageBody()));
+      await tester.pumpAndSettle();
+      check(find.byType(SubscriptionListPageBody)).findsOne();
```

Comment:
```suggestion
      await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
        child: const HomePage()));
      await tester.pump();
      await tester.tap(find.byIcon(ZulipIcons.hash_italic));
      await tester.pump();
      check(find.byType(SubscriptionListPageBody)).findsOne();
```

To be more representative of the real app.